src/Concise/Extensions/Pho/PhoTestSuite.php

```diff
@@ -0,0 +1,169 @@
+
+        // The diff is empty in case a parent class (with test methods) is added
+        // AFTER a child class that inherited from it. To account for that case,
+        // cumulate all discovered classes, so the parent class may be found in
+        // a later invocation.
+        if ($newClasses) {
+            // On the assumption that test classes are defined first in files,
+            // process discovered classes in approximate LIFO order, so as to
+            // avoid unnecessary reflection.
+            $this->foundClasses = array_merge($newClasses, $this->foundClasses);
+        }
+
+        // The test class's name must match the filename, either in full, or as
+        // a PEAR/PSR-0 prefixed shortname ('NameSpace_ShortName'), or as a
+        // PSR-1 local shortname ('NameSpace\ShortName'). The comparison must be
+        // anchored to prevent false-positive matches (e.g., 'OtherShortName').
+        $shortname = basename($filename, '.php');
+        $shortnameRegEx = '/(?:^|_|\\\\)' . preg_quote($shortname, '/') . '$/';
+
+        foreach ($this->foundClasses as $i => $className) {
+            if (preg_match($shortnameRegEx, $className)) {
+                $class = new ReflectionClass($className);
+
+                if ($class->getFileName() == $filename) {
+                    $newClasses = array($className);
+                    unset($this->foundClasses[$i]);
+                    break;
+                }
+            }
+        }
+
+        foreach ($newClasses as $className) {
+            $class = new ReflectionClass($className);
+
+            if (!$class->isAbstract()) {
+                if ($class->hasMethod(
+                    PHPUnit_Runner_BaseTestRunner::SUITE_METHODNAME
+                )
+                ) {
+                    $method = $class->getMethod(
+                        PHPUnit_Runner_BaseTestRunner::SUITE_METHODNAME
+                    );
+
+                    if ($method->isStatic()) {
+                        $this->addTest($method->invoke(null, $className));
+                    }
+                } elseif ($class->implementsInterface(
+                    'PHPUnit_Framework_Test'
+                )
+                ) {
+                    $this->addTestSuite($class);
+                }
+            }
+        }
+
+        $tokens = token_get_all(file_get_contents($filename));
+        PhoTestCase::$count += count(
+            array_filter(
+                $tokens,
+                function ($token) {
+                    return
+                        is_array($token) &&
+                        $token[0] == 308 &&
+                        $token[1] == 'it';
+                }
+            )
+        );
+
+        static $done = false;
+        if (!$done) {
+            $this->addTestSuite(
+                new \ReflectionClass('Concise\Extensions\Pho\PhoTestCase')
+            );
+            $done = true;
+        }
+    }
```

Comment:
Can this be chopped down to only whats needed, or at least documents why all this code needs to be here.